### PR TITLE
Convert tests to Typescript

### PR DESCRIPTION
**Pull request by jas14**
I'm working on #374 and am writing some unit tests. I can't get the current custom Jest preprocessor to work with TypeScript declaration files, so, figured I'd type the existing tests and use `ts-jest`.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ android:
       - extra-android-m2repository
 sudo: false
 before_install:
-  - nvm install 10.16.3
+  - nvm install 12.2.0
```
Comment (jas14) [inline .travis.yml:14]: Node 10 for some reason errored on `import 'whatwg-fetch'`, but Node 12 didn't. The README says the Node dependency is "latest LTS," which I guess is 14.x, but I'm hoping the latest 12.x is OK here.

File: tsconfig.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "noUnusedLocals": true,
     "jsx": "react",
     "jsxFactory": "h",
-    "types": ["lodash-es", "d3"]
+    "types": ["lodash-es", "d3", "jest"]
```
Comment (veloce) [inline tsconfig.json:18]: I'd like not to add jest types to the main tsconfig.json. It looks like it's possible to define a custom tsconfig for jest (that extends the global one) by modifying jest config like so:
```
{
  ...
  "globals": {
    "ts-jest": {"tsConfigFile": "tsconfig.test.json"}
  }
}
```

**Review by veloce — CHANGES_REQUESTED** [review]
Thanks! looks fine except for the tsconfig modif.

File: src/dts/index.d.ts [hunk 3]
```diff
@@ -6,3 +6,6 @@
 /// <reference path="signals.d.ts" />
 /// <reference path="misc.d.ts" />
 /// <reference path="mithril.d.ts" />
+
```
Comment (veloce) [inline src/dts/index.d.ts:9]: node types are not used so this should be removed

**Review by veloce — CHANGES_REQUESTED** [review]
just one more change, and I'll merge it

File: src/dts/index.d.ts (hunk 3 above)
Comment (jas14) [reply src/dts/index.d.ts]: Whoops. fixed

**Pull request merged**